docs/advanced/resilience-context.md

```diff
@@ -68,6 +68,37 @@ public static class MyResilienceKeys
 > [!NOTE]
 > We recommend defining a static class to hold the resilience property keys used in your project. This approach makes these keys easier to discover and maintain. For simpler scenarios, you can directly use the creation of `ResiliencePropertyKey<string>` since it's a cheap, struct-based API.
 
+### Sequence diagram
+
+```mermaid
+sequenceDiagram
+    autonumber
+    actor C as Caller
+    participant CP as ResilienceContextPool
+    participant P as Pipeline
+    participant R as Retry
+    participant D as DecoratedUserCallback
+    participant O as OnRetryUserCallback
+
+    C->>CP: Rents a context
+    CP->>C: Gives a context
+    C->>P: Calls ExecuteAsync<br/>with context
+    P->>R: Calls ExecuteCore<br/>with context
+    Note over R,D: Initial attempt
+    R->>+D: Invokes<br/>with context
+    D->>-R: Fails
+    R->>+O: Invokes<br/>with context
+    O->>-R: Returns a ValueTask
```

Comment:
Do we need this detail? This is more of an implementation detail as it's async, but otherwise it doesn't return anything. Maybe we just have the one line that's like "Awaits {the thing}".

Reply:
How about simply just replacing `Returns a ValueTask` to `Returns`/`Finishes`/`Completes`?